snuba/query/parser/functions.py

```diff
@@ -1,13 +1,96 @@
 import numbers
+import re
 from datetime import date, datetime
-from typing import Any, Callable, List, Optional, TypeVar, Tuple, Union
+from typing import Any, Callable, List, Optional, Tuple, TypeVar, Union
 
-from snuba.query.expressions import Expression, Literal, FunctionCall
+from sentry_relay.consts import SPAN_STATUS_NAME_TO_CODE
+
+from snuba.clickhouse.escaping import escape_identifier
+from snuba.query.expressions import Expression, FunctionCall, Literal
 from snuba.query.parser.strings import parse_string_to_expr
+from snuba.state import get_config
 from snuba.util import is_function
 
 TExpression = TypeVar("TExpression")
 
+TOPK_FUNCTION_RE = re.compile(r"^top([1-9]\d*)$")
+APDEX_FUNCTION_RE = re.compile(r"^apdex\(\s*([^,]+)+\s*,\s*([\d]+)+\s*\)$")
+IMPACT_FUNCTION_RE = re.compile(
+    r"^impact\(\s*([^,]+)+\s*,\s*([\d]+)+\s*,\s*([^,]+)+\s*\)$"
+)
+FAILURE_RATE_FUNCTION_RE = re.compile(r"^failure_rate\(\)$")
+
+
+def function_expr(fn: str, args_expr: str = "") -> str:
```

Comment:
This is just cut pasted from util.py. This is because I need to make it depend on snuba.state, and leaving it in util.py would cause a circular dependency.